src/javaCoreTwo/collectionsAndArrays/FifthPart.java

```diff
@@ -0,0 +1,65 @@
+package javaCoreTwo.collectionsAndArrays;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+public class FifthPart {
+    static class User {
+        private String name;
+        private int age;
+
+        public User(String name, int age) {
+            this.name = name;
+            this.age = age;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        public int getAge() {
+            return age;
+        }
+
+        public void setAge(int age) {
+            this.age = age;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if(this == obj)
+                return true;
+
+            if(obj == null || getClass() != obj.getClass())
+                return false;
+
+            User user = (User) obj;
+
+            return age == user.age && name.equals(user.name);
```

Comment:
return age == user.age && Objects.equals(name, user.name); would also compare fields and thus handle potential null values for the name field